vms/avm/vm.go

```diff
@@ -822,20 +812,21 @@ func (vm *VM) getUTXO(utxoID *avax.UTXOID) (*avax.UTXO, error) {
 		return utxo, nil
 	}
 
-	inputTx, inputIndex := utxoID.InputSource()
-	parent := UniqueTx{
-		vm:   vm,
-		txID: inputTx,
-	}
+	inputTxID, inputIndex := utxoID.InputSource()
+	parent := newUniqueTx(vm, inputTxID, nil)
 
-	// TODO do we need to call parent.verifyWithoutCacheWrites here?
+	// If the transaction has been accepted and the UTXO was
+	// not found above, then it has already been consumed.
+	// If the parent was rejected, the UTXO cannot be consumed.
+	// If the parent is unknown, then it is too early to consume
+	// this UTXO.
 	if status := parent.Status(); status != choices.Processing {
 		return nil, errMissingUTXO
 	}
 
 	parentUTXOs := parent.UTXOs()
 	if uint32(len(parentUTXOs)) <= inputIndex || int(inputIndex) < 0 {
-		return nil, errInvalidUTXO
+		return nil, fmt.Errorf("invalid UTXO index %d for parent with %d UTXOs", inputIndex, len(parentUTXOs))
```

Comment:
`fmt.Errorf("invalid output index %d for parent tx %s (%d outputs)", inputIndex, inputTxID, len(parentUTXOs))`